qmcpy/discrete_distribution/kronecker.py

```diff
@@ -0,0 +1,128 @@
+from ._discrete_distribution import LD
+from numpy import *
+from sympy import nextprime
+
+PRIMES = array([2,   3,   5,   7,  11,  13,  17,  19,  23,  29,  31,  37,  41, 
+                43,  47,  53,  59,  61,  67,  71,  73,  79,  83,  89,  97, 101,
+                103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167,
+                173, 179, 181, 191, 193, 197, 199, 211, 223, 227, 229, 233, 239,
+                241, 251, 257, 263, 269, 271, 277, 281, 283, 293, 307, 311, 313,
+                317, 331, 337, 347, 349, 353, 359, 367, 373, 379, 383, 389, 397,
+                401, 409, 419, 421, 431, 433, 439, 443, 449, 457, 461, 463, 467,
+                479, 487, 491, 499, 503, 509, 521, 523, 541])
+
+RICHTMYER = sqrt(PRIMES) % 1
+
+class Kronecker(LD):
+    def __init__(self, dimension=1, alpha=0, delta=0, replications=1, randomize=False, m_max=2 ** 20, seed=None):
+        # attributes required for cub_qmc_clt.py
+        self.mimics = 'StdUniform'
+        self.d = dimension
+        self.replications = replications
+        self.randomize = randomize
+        self.dimension = dimension
+        self.low_discrepancy = True
+        self.d_max = dimension
+        self.m_max = m_max
+        
+        if type(alpha) == str and alpha.lower() == 'richtmyer':
+                if dimension <= len(PRIMES):
+                    self.alpha = RICHTMYER[:dimension]
+                else:
+                    self.alpha = append(RICHTMYER, [sqrt(nextprime(PRIMES[-1], ith=x)) % 1 for x in range(1, dimension - len(PRIMES) + 1)])
+        else:
+            if sum(alpha) == 0:
+                self.alpha = random.rand(dimension)
+            else:
+                self.alpha = alpha
+
+        super(Kronecker,self).__init__(dimension, seed)
+
+        if self.randomize:
+            self.delta = random.rand(self.replications, self.dimension)
+        elif sum(delta) == 0:
+            if seed is None:
+                self.delta = zeros(dimension)
+            else:
+                self.delta = random.rand(dimension)
+        else:
+            self.delta = delta
+
+    def gen_samples(self, n=None, n_min=0, n_max=0):
+        # returns replications x (n_max-n_min) x d (dimension) array of samples
+        if n is None:
+            n = n_max - n_min
+
+        i = arange(n).reshape((n, 1))
+
+        if self.replications == 1:
+            points = (i * self.alpha) + self.delta
+            return subtract(points, floor(points), out = points)
+
+        samples = empty(shape=(self.replications, n, self.dimension))
+        for r in range(self.replications):
+            samples[r] = ((i * self.alpha) + self.delta[r]) % 1
+
+        return samples
+
+
+    def periodic_discrepancy(self, n, k_tilde=None, gamma=None):
```

Comment:
Leave the discrepancy functions for a separate PR. I don't think we want to them attached to a specific class as, e.g., these can be used for either lattices or Kronecker points. we will have to think more carefully about how to implement discrepancies into QMCPy